kong-ngx-build

```diff
@@ -387,6 +393,70 @@ main() {
   succ "Build finished in $SECONDS seconds. Enjoy!"
 }
 
+parse_version() {
+  [[ -z $1 || -z $2 ]] && return 1
+
+  # remove non-version parts of a version, ie foobar-1.2.3a.5-rc1 -> 1.2.3a.5
+  local re='^[^0-9]*([0-9a-zA-Z.]*).*$'
```

Comment:
Maybe just `^[^0-9]*([0-9a-zA-Z.]*)`?

Reply:
But then we do not discard the latter part of the version in `foo-1.2.3-5-rc19`. If we are to effectively parse `-rc19` in `5-rc19` then a couple of the regexes need to change.

Reply:
Maybe we are lacking some tests covering these cases?